public/app/plugins/datasource/tempo/streaming.ts

```diff
@@ -100,6 +105,116 @@ export function doTempoChannelStream(
+
+  return getGrafanaLiveSrv()
+    .getStream<MutableDataFrame>({
+      scope: LiveChannelScope.DataSource,
+      namespace: ds.uid,
+      path: `metrics/${key}`,
+      data: {
+        ...query,
+        step,
+        timeRange: {
+          from: range.from.toISOString(),
+          to: range.to.toISOString(),
+        },
+      },
+    })
+    .pipe(
+      takeWhile((evt) => {
+        if ('message' in evt && evt?.message) {
+          const frameState: SearchStreamingState = evt.message.data.values[2][0];
+          if (frameState === SearchStreamingState.Done || frameState === SearchStreamingState.Error) {
+            return false;
+          }
+        }
+        return true;
+      }, true),
+      map((evt) => {
+        let newResult: DataQueryResponse = { data: [], state: LoadingState.NotStarted };
+        if ('message' in evt && evt?.message) {
+          // Schema should be [data, metrics, state, error]
+          const data = evt.message.data.values[0][0];
+          const frameState: SearchStreamingState = evt.message.data.values[2][0];
+          const error = evt.message.data.values[3][0];
+
+          switch (frameState) {
+            case SearchStreamingState.Done:
+              state = LoadingState.Done;
+              break;
+            case SearchStreamingState.Streaming:
+              state = LoadingState.Streaming;
+              break;
+            case SearchStreamingState.Error:
+              throw new Error(error);
+          }
+
+          newResult = {
+            data: data?.map(dataFrameFromJSON) ?? [],
+            state,
+          };
+        }
+
+        return newResult;
+      }),
+      // Merge results on acc
+      scan((acc, curr) => {
+        if (!curr) {
+          return acc;
+        }
+        if (!acc) {
+          return cloneQueryResponse(curr);
+        }
+        return mergeFrames(acc, curr);
+      })
+    );
+}
+
+function mergeFrames(acc: DataQueryResponse, newResult: DataQueryResponse): DataQueryResponse {
+  const result = combineResponses(cloneQueryResponse(acc), newResult);
+
+  // Remove duplicate time field values for all frames
+  result.data.forEach((frame: DataFrame) => {
+    const timeFieldIndex = frame.fields.findIndex((f) => f.type === FieldType.time);
+    if (timeFieldIndex >= 0) {
+      const timeField = frame.fields[timeFieldIndex];
+      removeDuplicateTimeFieldValues(frame, timeField);
+      sortDataFrame(frame, timeFieldIndex);
+    }
+  });
+
+  result.state = newResult.state;
+  return result;
+}
+
+function removeDuplicateTimeFieldValues(accFrame: DataFrame, timeField: Field) {
+  const duplicatesMap = timeField.values.reduce((acc: Record<number, number[]>, value, index) => {
+    if (acc[value]) {
+      acc[value].push(index);
+    } else {
+      acc[value] = [index];
+    }
+    return acc;
+  }, {});
+
+  const indexesToRemove = Object.values(duplicatesMap)
+    .filter((indexes) => indexes.length > 1)
+    .map((indexes) => indexes[1]);
```

Comment:
It works fine if there's only one duplicate but IIUC this will remove only one duplicate if there's more than 1, e.g. having

| Time | Value |
|--------|--------|
| 100 | 0.4 |
| 101 | 0.1 |
| 101 | 0.2 |
| 101 | 0.3 |
| 102 | 0.5 |

It would filter out the second 101 only:

| Time | Value |
|--------|--------|
| 100 | 0.4 |
| 101 | 0.1 |
| 101 | 0.3 |
| 102 | 0.5 |

Do we need to cover such case? Should there be a unit test to cover it? 🤔